after-effects/src/aegp/suites/persistent_data.rs

```diff
@@ -0,0 +1,502 @@
+    pub fn num_keys(
+        &self,
+        blob_handle: PersistentBlobHandle,
+        section_key: &str,
+    ) -> Result<i32, Error> {
+        let section_key = CString::new(section_key).map_err(|_| Error::InvalidParms)?;
+        call_suite_fn_single!(self, AEGP_GetNumKeys -> A_long, blob_handle.into(), section_key.as_ptr())
+    }
+
+    /// Retrieves the value of the indexed key.
+    pub fn value_by_key_index(
+        &self,
+        blob_handle: PersistentBlobHandle,
+        section_key: &str,
+        key_index: i32,
+        max_key_size: usize,
+    ) -> Result<String, Error> {
+        let section_key = CString::new(section_key).map_err(|_| Error::InvalidParms)?;
+        let mut out_data = vec![0i8; max_key_size + 1];
+
+        call_suite_fn!(
+            self,
+            AEGP_GetValueKeyByIndex,
+            blob_handle.into(),
+            section_key.as_ptr(),
+            key_index,
+            max_key_size as _,
+            out_data.as_mut_ptr() as _,
+        )?;
+
+        buffer_to_string(out_data)
+    }
+
+    pub fn data_handle<'a, T>(
+        &self,
+        plugin_id: PluginId,
+        blob_handle: PersistentBlobHandle,
+        section_key: &str,
+        value_key: &str,
+        default: MemHandle<'a, T>,
+    ) -> Result<MemHandle<T>, Error> {
+        let section_key = CString::new(section_key).map_err(|_| Error::InvalidParms)?;
+        let value_key = CString::new(value_key).map_err(|_| Error::InvalidParms)?;
+
+        let handle = call_suite_fn_single!(self, AEGP_GetDataHandle -> ae_sys::AEGP_MemHandle,
+                                           plugin_id,
+                                           blob_handle.as_ptr(),
+                                           section_key.as_ptr(),
+                                           value_key.as_ptr(),
+                                           default.as_raw())?;
+
+        MemHandle::from_raw(handle)
+    }
+
+    /// Obtains the data located at a given section's value.
+    pub fn data(
+        &self,
+        blob_handle: PersistentBlobHandle,
+        section_key: &str,
+        value_key: &str,
+        data_size: u32,
+        default: *const c_void,
+    ) -> Result<*mut c_void, Error> {
+        let section_key = CString::new(section_key).map_err(|_| Error::InvalidParms)?;
+        let value_key = CString::new(value_key).map_err(|_| Error::InvalidParms)?;
+
+        let ptr = std::ptr::null_mut();
+
+        call_suite_fn!(
+            self,
+            AEGP_GetData,
+            blob_handle.as_ptr(),
+            section_key.as_ptr(),
+            value_key.as_ptr(),
+            data_size,
+            default,
+            ptr
+        )?;
+
+        Ok(ptr)
+    }
+
+    /// Obtains the string for a given section key's value
+    ///
+    /// Note: This interperets all stored strings as UTF-8. This is safe if you are retrieving
+    /// strings stored from rust, which must be UTF-8.
+    pub fn string(
```

Comment:
I stuck with the no `get_` prefix naming convention but granted that these are only type names if following the convention we might make an exception here.